algorithms/src/signature/schnorr.rs

```diff
@@ -166,32 +173,53 @@ where
     ) -> Result<Self::Signature, SignatureError> {
         let sign_time = start_timer!(|| "SchnorrSignature::sign");
         // (k, e);
-        let (random_scalar, verifier_challenge) = loop {
-            // Sample a random scalar `k` from the prime scalar field.
-            let random_scalar: <G as Group>::ScalarField = <G as Group>::ScalarField::rand(rng);
-            // Commit to the random scalar via r := k · g.
-            // This is the prover's first msg in the Sigma protocol.
-            let mut prover_commitment = G::zero();
-            for (bit, base_power) in
-                from_bytes_le_to_bits_le(&to_bytes![random_scalar]?).zip_eq(&self.parameters.generator_powers)
-            {
-                if bit {
-                    prover_commitment += base_power;
-                }
-            }
 
-            // Hash everything to get verifier challenge.
-            let mut hash_input = Vec::new();
-            hash_input.extend_from_slice(&self.parameters.salt);
-            hash_input.extend_from_slice(&to_bytes![prover_commitment]?);
-            hash_input.extend_from_slice(message);
+        // Sample a random scalar `k` from the prime scalar field.
+        let random_scalar: <G as Group>::ScalarField = <G as Group>::ScalarField::rand(rng);
+        // Commit to the random scalar via r := k · g.
+        // This is the prover's first msg in the Sigma protocol.
+        let mut prover_commitment = G::zero();
+        for (bit, base_power) in
+            from_bytes_le_to_bits_le(&to_bytes![random_scalar]?).zip_eq(&self.parameters.generator_powers)
+        {
+            if bit {
+                prover_commitment += base_power;
+            }
+        }
 
-            // Compute the supposed verifier response: e := H(salt || r || msg);
-            if let Some(verifier_challenge) = <G as Group>::ScalarField::from_random_bytes(&D::digest(&hash_input)) {
-                break (random_scalar, verifier_challenge);
-            };
+        // Hash everything to get verifier challenge.
+        let mut hash_input = Vec::<<G as AffineCurve>::BaseField>::new();
+        hash_input.extend_from_slice(&self.parameters.salt.to_field_elements().unwrap());
```

Comment:
@raychu86 heads up that the branch with replacement of `salt` to `public_key` will conflict with the changes here.

Reply:
That should be doable, as the public key will be a point, which can be hashed into the sponge just like the "commitment" in the next line.

Reply:
@raychu86 Do you want which branch it is? If the change is small, I can merge it into this PR.